pkgs/development/interpreters/python/mk-python-derivation.nix

```
@@ -303,6 +305,9 @@ let
         ++ optionals (attrs ? pythonRelaxDeps || attrs ? pythonRemoveDeps) [
           pythonRelaxDepsHook
         ]
+        ++ optionals relaxBuildDeps [
+          pythonRelaxBuildDepsHook
```

Dynamically toggling `__structuredAttrs` based on the relaxing attributes would be very surprising. I prefer throwing an error if the hook is invoked an `__structuredAttrs` is false.

```suggestion
          (
            lib.throwIf
              (! finalAttrs.__structuredAttrs or false)
              "${finalAttrs.pname or finalAttrs.name}: pythonRelaxBuildDepsHook requires __structuredAttrs = true."
              pythonRelaxBuildDepsHook
          )
```

---

Fair.  :+1: